[demo] rename demo services (#6438)
open-telemetry/opentelemetry.io: content/en/docs/demo/requirements/architecture.md

Each service should be interchangeable with a service that performs the same
business logic, implementing the same gRPC endpoints, but written in a different
language/implementation.

Each service should be able to communicate with a feature flag service in order
to enable/disable faults that can be used to illustrate how telemetry helps
solve problems in distributed applications.

## Feature Flag Component

Feature flagging is a crucial part of cloud native application development. The

> Demo uses OpenFeature, a CNCF incubating project, to manage feature flags.

Reviewer: ```suggestion
demo uses OpenFeature, a CNCF incubating project, to manage feature flags.
```

> A PHP service should be added to the main application as an 'admin service'. A

Reviewer: ```suggestion
A PHP service should be added to the main application as an admin service. A
```

Database should be added to enable CRUD functionality on the Product Catalog.

The 'shipping' service should be reimplemented in Rust.

> The 'currency' service should be reimplemented in C++.

Reviewer: ```suggestion
The currency service should be reimplemented in C++.
```

> The 'email' service should be reimplemented in Ruby.

Reviewer: ```suggestion
The email service should be reimplemented in Ruby.
```
